clients/ts/FunctionalTests/ts/HubConnectionTests.ts

```diff
@@ -544,6 +544,32 @@ describe("hubConnection", () => {
         }
     });
 
+    it("transport falls back from WebSockets to SSE or LongPolling", async (done) => {
+        // Replace Websockets with a function that just
+        // throws to force fallback.
+        const oldWebSocket = (window as any).WebSocket;
+        (window as any).WebSocket = () => {
+            throw new Error("Kick rocks");
+        };
+
+        const hubConnection = new HubConnection(TESTHUBENDPOINT_URL, {
+            logger: LogLevel.Trace,
+            protocol: new JsonHubProtocol(),
+        });
+
+        try {
+            await hubConnection.start();
+
+            // Make sure that we connect with SSE or LongPolling after Websockets fail
+            const transportName = await hubConnection.invoke("GetActiveTransportName");
+            expect(transportName === "ServerSentEvents" || transportName === "LongPolling").toBe(true);
+            (window as any).WebSocket = oldWebSocket;
```

Comment:
Put this in a `finally` block. If the test throws, we don't want to leave `window.WebSocket` in a broken state.